### PR TITLE
[NO-JIRA]BpkInput to forwardRef

### PR DESCRIPTION
**Pull request by LouiseReid**
Draft PR to show changes required to update component to a forward ref. Full contribution will also update this to typescript

Remember to include the following changes:

- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [ ] Tests
- [ ] Storybook examples created/updated

### CODE REVIEW

File: packages/bpk-component-input/src/BpkInput.js [hunk 1]
```diff
@@ -128,12 +111,7 @@ class BpkInput extends Component<Props, State> {
       // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
       <input
         className={classNames.join(' ')}
-        ref={(input) => {
-          ref = input;
-          if (inputRef) {
-            inputRef(input);
-          }
-        }}
+        ref={forwardedRef}
```
Comment (anambl) [inline packages/bpk-component-input/src/BpkInput.js:114]: I guess we'd still need the component to work even if there is no ref passed as `inputRef` is an optional prop?

Comment (LouiseReid) [reply packages/bpk-component-input/src/BpkInput.js:114]: I think with forward refs if you pass no ref it just handle it as null so its not the same as not sending a prop. Let me check

Comment (LouiseReid) [reply packages/bpk-component-input/src/BpkInput.js:114]: Yea it just passes it as null - https://codesandbox.io/s/react-playground-forked-s7k7cd?file=/index.js

**Comment by anambl** [comment]
Hey @LouiseReid can I close this for now? 🙂

**Comment by LouiseReid** [comment]
> Hey @LouiseReid can I close this for now? 🙂

Yep sorry forgot this was still here

**Pull request closed without merging**